File: scripts/hooks/doc-file-warning.js
#!/usr/bin/env node
/**
 * doc-file-warning.js — Warn about non-standard documentation files
 *
 * Advisory (exit 0). Warns when creating documentation files outside
 * recognized paths to keep project structure clean.
 */

const fs = require('fs');
const path = require('path');

const RECOGNIZED_DOCS = [
  'README.md', 'CONTRIBUTING.md', 'CHANGELOG.md', 'SECURITY.md',
  'LICENSE', 'CLAUDE.md', 'AGENTS.md', 'ROADMAP.md',
  'CODE_OF_CONDUCT.md', 'ARCHITECTURE.md'
];

const RECOGNIZED_DIRS = [
  'docs/', 'skills/', 'agents/', 'commands/', '.opc/'
];

try {
  const raw = fs.readFileSync(0, 'utf8');
  const input = JSON.parse(raw);
  const filePath = (input.tool_input && (input.tool_input.file_path || input.tool_input.path)) || '';

  if (!filePath || !filePath.endsWith('.md')) {
    process.exit(0);
  }

  const basename = path.basename(filePath);
  const isRecognized = RECOGNIZED_DOCS.includes(basename) ||
    RECOGNIZED_DIRS.some(dir => filePath.includes(dir));

  if (!isRecognized) {
    console.log(
      'SuperOPC: Creating "' + basename + '" outside standard paths. ' +
      'Consider placing docs in docs/, skills in skills/, or using a recognized root file.'
    );
  }
} catch (e) {
  // Silent pass
}

process.exit(0);

File: scripts/hooks/doc_file_warning.py
#!/usr/bin/env python3

from __future__ import annotations

from pathlib import Path
import sys

from common import get_first_path, get_tool_input, read_stdin_json, write_message


RECOGNIZED_DOCS = {
    "README.md",
    "CONTRIBUTING.md",
    "CHANGELOG.md",
    "SECURITY.md",
    "LICENSE",
    "CLAUDE.md",
    "AGENTS.md",
    "ROADMAP.md",
    "CODE_OF_CONDUCT.md",
    "ARCHITECTURE.md",
}
RECOGNIZED_DIRS = ("docs/", "skills/", "agents/", "commands/", ".opc/")


def main() -> int:
    payload = read_stdin_json()
    file_path = get_first_path(get_tool_input(payload))
    if not file_path or not file_path.endswith(".md"):
        return 0

    basename = Path(file_path).name
    normalized = file_path.replace("\\", "/")
    is_recognized = basename in RECOGNIZED_DOCS or any(part in normalized for part in RECOGNIZED_DIRS)
    if not is_recognized:
        write_message(
            f'SuperOPC: Creating "{basename}" outside standard paths. '
            "Consider placing docs in docs/, skills in skills/, or using a recognized root file."
        )

    return 0


if __name__ == "__main__":
    sys.exit(main())
